### PR TITLE
feat: refund confirmation page API

### PR DESCRIPTION
**Pull request by NirmalHegde**
## Notion ticket link

[Ticket Name](https://uwblueprintexecs.notion.site/Refund-Confirmation-API-20bc2f90b63b4e6194ac8a00d8723730)

## Implementation description
* Implemented get refund confirmation to get info of user wanting to get refund.
* Provided direct link access for server

## Steps to test
1. ensure that 404, 500, and 200 result paths are reached
2. ensure right data comes back

## What should reviewers focus on?
* ensure 200 path provides correct info

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/rest/camperRoutes.ts [hunk 1]
```diff
@@ -76,4 +76,20 @@ camperRouter.get("/", async (req, res) => {
   }
 });
 
+camperRouter.get("/refund-confirm/:chargeId", async (req, res) => {
+  const { chargeId } = req.params;
+  try {
+    const camper = await camperService.getCamperByChargeId(
+      (chargeId as unknown) as number,
```
Comment (annechung) [inline backend/typescript/rest/camperRoutes.ts:83]: chargeId was updated to a string type

File: backend/typescript/services/implementations/camperService.ts [hunk 2]
```diff
@@ -167,6 +167,41 @@ class CamperService implements ICamperService {
 
     return camperDtos;
   }
+
+  async getCamperByChargeId(chargeId: number): Promise<CamperDTO> {
```
Comment (annechung) [inline backend/typescript/services/implementations/camperService.ts:171]:
```suggestion
  async getCamperByChargeId(chargeId: string): Promise<CamperDTO> {
```

File: backend/typescript/services/interfaces/camperService.ts [hunk 3]
```diff
@@ -23,6 +23,14 @@ interface ICamperService {
...
    */
   getCampersByCampId(campId: string): Promise<Array<CamperDTO>>;
+
+  /**
+   * Get campers associated with charge id
+   * @param chargeId camper's charge id for refunds
+   * @returns CamperDTO
+   * @throws Error if camper retrieval fails
+   */
+  getCamperByChargeId(chargeId: number): Promise<CamperDTO>;
```
Comment (annechung) [inline backend/typescript/services/interfaces/camperService.ts:33]:
```suggestion
  getCamperByChargeId(chargeId: string): Promise<CamperDTO>;
```

File: backend/typescript/services/implementations/camperService.ts [hunk 4]
```diff
@@ -167,6 +167,41 @@ class CamperService implements ICamperService {
...
+    try {
+      // eslint-disable-next-line prettier/prettier
+      const camper: Camper | null = await MgCamper.findOne({ "chargeId": chargeId });
```
Comment (annechung) [inline backend/typescript/services/implementations/camperService.ts:174]: there can be multiple campers with the same chargeId and we should be returning all of them

File: backend/typescript/services/implementations/camperService.ts [hunk 5]
```diff
@@ -167,6 +167,41 @@ class CamperService implements ICamperService {
...
+
+      if (!camper) {
+        throw new Error(`Camper with Charge Id ${chargeId} not found.`);
+      }
+
+      const camperDTO: CamperDTO = {
```
Comment (annechung) [inline backend/typescript/services/implementations/camperService.ts:180]: should update this to use the newer CamperDTO

File: backend/typescript/rest/camperRoutes.ts [hunk 6]
```diff
@@ -76,4 +76,20 @@ camperRouter.get("/", async (req, res) => {
...
+    );
+    res.status(200).json(camper);
+  } catch (error: unknown) {
+    if (getErrorMessage(error).includes("Camper with Charge Id")) {
```
Comment (annechung) [inline backend/typescript/rest/camperRoutes.ts:87]: this works but is a bit jank :'D would it be possible to use the cause property on the Error object before throwing it in the camper service and check that property here? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error

**Review by annechung — COMMENTED** [review]
good start, but a few changes have to be made to return multiple campers and also to use the latest dto!

File: backend/typescript/rest/camperRoutes.ts (hunk 6 above)
Comment (NirmalHegde) [reply backend/typescript/rest/camperRoutes.ts:87]: for sure!

Comment (annechung) [reply backend/typescript/rest/camperRoutes.ts]: returning just the one 500 error works tooo, we can leave it as it

**Review by annechung — APPROVED** [review]
tested and looks good!! can merge this in :D

**Pull request merged**